tools/txg/TXG/Types.hs

```diff
@@ -83,118 +75,134 @@ data TimingDistribution
 
-pollkeys :: A.Parser TransactionCommand
+pollkeys :: A.Parser TXCmd
 pollkeys = do
   _constructor <- A.string "poll"
   A.skipSpace
   _open <- A.char '[' >> A.skipSpace
   bs <- parseRequestKey
   _close <- A.skipSpace >> A.char ']'
-  A.skipSpace
-  measure <- MeasureTime <$> ((False <$ A.string "false") <|> (True <$ A.string "true"))
-  pure $ PollRequestKeys bs measure
+  pure $ PollRequestKeys bs
 
 parseRequestKey :: A.Parser ByteString
 parseRequestKey = B8.pack <$> A.count 128 (A.satisfy (A.inClass "abcdef0123456789"))
 
-listenkeys :: A.Parser TransactionCommand
+listenkeys :: A.Parser TXCmd
 listenkeys = do
   _constructor <- A.string "listen"
   A.skipSpace
   bytestring <- parseRequestKey
-  A.skipSpace
-  measure <- MeasureTime <$> ((False <$ A.string "false") <|> (True <$ A.string "true"))
-  pure $ ListenerRequestKey bytestring measure
+  pure $ ListenerRequestKey bytestring
 
-instance HasTextRepresentation TransactionCommand where
+instance HasTextRepresentation TXCmd where
   toText = transactionCommandToText
   {-# INLINE toText #-}
   fromText = transactionCommandFromText
   {-# INLINE fromText #-}
 
-instance Default TransactionCommand where
-  def = RunSimpleExpressions def def
+instance Default TXCmd where
+  def = RunSimpleExpressions def
 
----------------
--- ScriptConfig
----------------
+-------
+-- Args
+-------
 
-data ScriptConfig = ScriptConfig
-  { _scriptCommand :: !TransactionCommand
+data Args = Args
+  { _scriptCommand :: !TXCmd
   , _nodeChainIds :: ![ChainId]
   , _isChainweb :: !Bool
   , _hostAddresses :: ![HostAddress]
   , _nodeVersion :: !ChainwebVersion
-  , _logHandleConfig :: !U.HandleConfig }
-  deriving (Show, Generic)
-
-makeLenses ''ScriptConfig
-
-instance ToJSON ScriptConfig where
-  toJSON o =
-    object
-      [ "scriptCommand"       .= _scriptCommand o
-      , "nodeChainIds"        .= _nodeChainIds o
-      , "isChainweb"          .= _isChainweb o
-      , "hostAddresses"       .= _hostAddresses o
-      , "chainwebVersion"     .= _nodeVersion o
-      , "logHandle"           .= _logHandleConfig o
-      ]
-
-instance FromJSON (ScriptConfig -> ScriptConfig) where
-  parseJSON = withObject "ScriptConfig" $ \o -> id
-    <$< scriptCommand       ..: "scriptCommand"       % o
-    <*< nodeChainIds        ..: "nodeChainIds"        % o
-    <*< isChainweb          ..: "isChainweb"          % o
-    <*< hostAddresses       ..: "hostAddresses"       % o
-    <*< nodeVersion         ..: "chainwebVersion"     % o
-    <*< logHandleConfig     ..: "logging"             % o
-
-defaultScriptConfig :: ScriptConfig
-defaultScriptConfig = ScriptConfig
-  { _scriptCommand   = RunSimpleExpressions def def
+  , _logHandleConfig :: !U.HandleConfig
+  , _batchSize :: !BatchSize
+  } deriving (Show, Generic)
+
+scriptCommand :: Functor f => (TXCmd -> f TXCmd) -> Args -> f Args
+scriptCommand f sc = (\tc -> sc { _scriptCommand = tc }) <$> f (_scriptCommand sc)
```

Comment:
Why not use TH to generate Lenses?

Reply:
That previously was the case. TH caused an annoyance during this round of changes, so I removed it as an experiment.